src/psyclone/psyir/frontend/fparser2.py

```diff
@@ -105,97 +151,114 @@ def _check_bound_is_full_extent(array, dim, index, operator):
-        assert literal.datatype == DataType.INTEGER, \
-            "Expecting integer datatype but found '{0}'." \
-            "".format(literal.datatype)
-        assert literal.value == str(dim), \
-            "Expecting literal value '{0}' to be the same as the " \
-            "current array dimension '{1}'.".format(literal.value,
-                                                    str(dim))
-    except AssertionError as excinfo:
-        raise NotImplementedError(
-            "psyir/frontend/fparser2.py:_check_bound_is_full_extent():"
-            + str(excinfo))
-
-
-def _check_literal(node, dim, index, value):
-    '''Utility function to check that the supplied array has a literal at
-    dimension index "dim" and range index "index" with value
-    "value".
+    _check_args(array, dim)
+
+    if operator == BinaryOperation.Operator.LBOUND:
+        index = 0
+    elif operator == BinaryOperation.Operator.UBOUND:
+        index = 1
+    else:
+        raise TypeError(
+            "'operator' argument  expected to be LBOUND or UBOUND but "
+            "found '{0}'.".format(type(operator).__name__))
+
+    # The first child of the array (index 0) relates to the first
+    # dimension (dim 1), so we need to reduce dim by 1.
+    bound = array.children[dim-1].children[index]
+
+    if not isinstance(bound, BinaryOperation):
+        return False
+
+    reference = bound.children[0]
+    literal = bound.children[1]
+
+    if (bound.operator == operator
+            and isinstance(reference, Reference) and
+            reference.symbol is array.symbol
+            and isinstance(literal, Literal) and
+            literal.datatype == DataType.INTEGER
+            and literal.value == str(dim)):
+        return True
+    return False
+
+
+def _is_array_range_literal(array, dim, index, value):
+    '''Utility function to check that the supplied array has an integer
+    literal at dimension index "dim" and range index "index" with
+    value "value".
 
     The step part of the range node has an integer literal with
     value 1 by default.
 
-    Assumes that the node argument is an array and that the
-    supplied dimension index is a Range node and that the supplied
-    range index is valid.
+    This routine is only in fparser2.py until #717 is complete as it
+    is used to check that array syntax in a where statement is for the
+    full extent of the dimension. Once #717 is complete this routine
+    can be moved into fparser2_test.py as it is used there in a
+    different context.
 
     :param array: the node to check.
     :type array: :py:class:`pysclone.psyir.node.Array`
     :param int dim: the dimension index to check.
     :param int index: the index of the range to check (0 is the \
-        lower bound, 1 is the upper bound).
+        lower bound, 1 is the upper bound and 2 is the step).
     :param int value: the expected value of the literal.
 
     :raises NotImplementedError: if the supplied argument does not \
```

Comment:
Please update and document the :return: and :rtype: as well.

Reply:
Missed this too. I also needed to update the exceptions.